config/plugin_selector_test.go

```diff
@@ -0,0 +1,203 @@
+package config
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestSetPluginLabelSelections(t *testing.T) {
+	tests := []struct {
+		name           string
+		selections     []string
+		expectedGroups []int // denotes the length of each group
+		wantErr        bool
+	}{
+		{
+			name:           "single selectors",
+			selections:     []string{"env=prod", "region=dc-23"},
+			expectedGroups: []int{1, 1}, // two groups, each with one selector
+		},
+		{
+			name:           "multiple selectors",
+			selections:     []string{"env=prod;region=dc-23", "env=dev;app=backend;policy=web"},
+			expectedGroups: []int{2, 3}, // two groups, one with two selectors and one with three
+		},
+		{
+			name:       "invalid selector syntax",
+			selections: []string{"env=prod;region=dc-23", "invalid-selector"},
+			wantErr:    true,
+		},
+		{
+			name:           "nil selectors",
+			selections:     nil,
+			expectedGroups: []int{0},
+		},
+		{
+			name:           "empty selector",
+			selections:     []string{""},
+			expectedGroups: []int{0},
+		},
+		{
+			name:           "multiple empty selectors",
+			selections:     []string{"", "app=web;env=prod*", ""},
+			expectedGroups: []int{2}, // only one valid group with 2 selectors
+		},
+		{
+			name:       "duplicate within group",
+			selections: []string{"env=prod;app=web;env=staging"},
+			wantErr:    true,
+		},
+		{
+			name:       "invalid key",
+			selections: []string{"invalid$key=prod", "region=dc-23"},
+			wantErr:    true,
+		},
+		{
+			name:       "invalid value",
+			selections: []string{"env=prod;app=web;invalid=value&()"},
+			wantErr:    true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			var pluginSelector labelSelector
+			err := pluginSelector.setSelections(tt.selections)
+			if tt.wantErr {
+				require.Error(t, err)
+				return
+			}
+			require.NoError(t, err)
+			for i, group := range pluginSelector.groups {
+				require.Equal(t, len(group), tt.expectedGroups[i])
+			}
+		})
+	}
+}
+
+func TestMatches(t *testing.T) {
+	tests := []struct {
+		name      string
+		selectors []string
+		labels    map[string]string
+		want      bool
```

Comment:
Please name this `expected` to be consistent with other tests in Telegraf...

Reply:
done